setup.py

```diff
@@ -223,6 +223,39 @@ def run(self):
             sys.exit(128)
 
 
+class Plugin(SimpleCommand):
+    """Handle plugins"""
+
+    description = 'Handle plugins'
+    user_options = [
+        ("list", 'l', "List available plugins"),
+        ("install=", 'i', "Plugin to install")
+    ]
+
+    def initialize_options(self):
+        self.list = False  # pylint: disable=W0201
+        self.install = None  # pylint: disable=W0201
+
+    def run(self):
+
+        plugins_list = []
+        directory = os.path.join(BASE_PATH, "optional_plugins")
+        for plugin in list(Path(directory).glob("*/setup.py")):
+            plugins_list.append(plugin.parts[-2])
+
+        if self.list or (not self.list and not self.install):
+            print("List of available plugins:\n ", "\n  ".join(plugins_list))
+            return
+
+        if self.install in plugins_list:
+            action = ["install", "--user"]
+            parent_dir = os.path.join(directory, self.install)
+            run([sys.executable, "setup.py"] + action, cwd=parent_dir, check=True)
```

Comment:
With the current method,  we support `requirements.txt` inside each plugin, if any requirement is defined there, they will be installed. According to the documentation `install_requires` keyword is supposed  to be minimal and has some limitations and might not cover all the cases.

With this change, what is the plan?

Reply:
My understanding is` install_requires` has all the requirements you need to run the plugin and you need `requirements.txt` for development. In the case of avocado optional plugin's, given it's the same base code, I'd expect to have all the plugins requirements for development in the main `requirements.txt' file.

Reply:
Unless I'm missing something, `requirements.txt` is not only for development. It can be used to specify *any* known-working environment  (i.e: dev, test, prod).

My only concern is this note on the documentation about install_requires:

> It is not considered best practice to use install_requires to pin dependencies to specific versions, or to specify sub-dependencies (i.e. dependencies of your dependencies). This is overly-restrictive, and prevents the user from gaining the benefit of dependency upgrades.
>
> Lastly, it’s important to understand that install_requires is a listing of “Abstract” requirements, i.e just names and version restrictions that don’t determine where the dependencies will be fulfilled from (i.e. from what index or source). 

About your comment:

> In the case of avocado optional plugin's, given it's the same base code, I'd expect to have all the plugins requirements for development in the main `requirements.txt' file.

TBH, I do have mix-feelings here. For instance, if someone is hacking the list plugin, not sure if makes sense to install all other requirements (specially if they are optional).

Anyway,  I just would  like to confirm the plans, since we don't have  requirements today, probably we could postpone  this discussion, and I don't see as blocking for this PR.